File: 1353_Maximum_Number_of_Events_That_Can_Be_Attended.py
import heapq

class Solution(object):
    def maxEvents(self, events):
        """
        :type events: List[List[int]]
        :rtype: int
        """
        events.sort(key=lambda x: x[1])

        attended_days = set()
        for start, end in events:
            for day in range(start, end + 1):
                if day not in attended_days:
                    attended_days.add(day)
                    break
        return len(attended_days)

File: 1353_Maximum_Number_of_Events_That_Can_Be_Attended.java
import java.util.Arrays;
import java.util.PriorityQueue;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int maxEvents(int[][] events) {
        // Sort events by end day to prioritize earlier finishing events
        Arrays.sort(events, (a, b) -> a[1] - b[1]);

        // Use a Set to keep track of days attended
        Set<Integer> daysAttended = new HashSet<>();
        for (int[] event : events) {
            for (int day = event[0]; day <= event[1]; day++) {
                if (!daysAttended.contains(day)) {
                    daysAttended.add(day);
                    break;
                }
            }
        }
        return daysAttended.size();
    }
}
